src/Build/Graph/GraphBuilder.cs

```diff
@@ -557,8 +579,15 @@ private List<ProjectInterpretation.ReferenceInfo> ParseReferences(ProjectGraphNo
                         referenceInfo.ReferenceConfiguration.ProjectFullPath
                         ));
                 }
-                
-                SubmitProjectForParsing(referenceInfo.ReferenceConfiguration);
+
+                if (!referenceInfo.ProjectReferenceItem.HasMetadata("SetPlatform"))
```

Comment:
We'll need to reconsider this logic when moving `DynamicPlatformResolution` to a global var in this class. I'm thinking we can add some bool `HadSetPlatformMetadataSet` (terrible name) to `ConfigurationMetadata` so that `ParseProject` knows when to skip the platform negotiation logic.

Reply:
moved to global var and reworked this.